src/js/components/projects/steps.tsx

```diff
@@ -0,0 +1,50 @@
+import i18n from 'i18next';
+import React from 'react';
+
+import Steps from '@/components/steps';
+import { Project } from '@/store/projects/reducer';
+import { Task } from '@/store/tasks/reducer';
+import { PROJECT_STATUSES } from '@/utils/constants';
+
+interface ProjectStatusStepsProps {
+  project: Project;
+  tasks: Task[];
+}
+
+const ProjectStatusSteps = ({ project, tasks }: ProjectStatusStepsProps) => {
+  const hasTasks = Boolean(tasks.length);
+  const hasDev = tasks.some((task) => task.assigned_dev);
+  const isMerged = project.status === PROJECT_STATUSES.MERGED;
+
+  const steps = [
+    {
+      label: `${i18n.t('Create a task')}`,
+      active: !hasTasks,
+      complete: hasTasks,
+    },
+    {
+      label: `${i18n.t('Assign a Developer to a task')}`,
+      active: hasTasks && !hasDev,
+      complete: project.has_unmerged_commits || (hasTasks && hasDev),
+    },
+    {
+      label: `${i18n.t('Complete a task')}`,
+      active: hasTasks && hasDev,
+      complete: project.has_unmerged_commits,
```

Comment:
@wlonk I'm realizing some discomfort for how we're handling `project.has_unmerged_commits`. Right now it is only set on project-creation (if using an existing branch), and when a task is merged. I'm trying to think of a good reason not to also update it when we receive a GH commit webhook reflecting that a commit was made to that project branch directly on GH? Then we could trust that it would more accurately reflect the state of the project branch...

Reply:
For me, given the capacity for history rewriting in git, that doesn't actually improve my trust much; only fully modeling the git commit graph could make me feel reasonably confident. That's not a reason to not make that change, but it's why I feel like it's moving from one approximation to another.

Reply:
That makes sense. I'm thinking about whether it would at least handle a few additional use-cases though... But maybe not worth it. The practical difference would just be whether users can submit a Metecho project for review (e.g. create a PR) if it contains commits made from outside Metecho. But as I articulate it, I'm not sure that's a valuable use-case to support in any case.